libarchive/archive_read_disk_entry_from_file.c

```diff
@@ -659,8 +675,12 @@ translate_acl(struct archive_read_disk *a,
 		// non-NFSv4 ACLs
 		entry_acl_type = default_entry_acl_type;
 #ifdef ACL_TYPE_NFS4
-		r = acl_get_entry_type_np(acl_entry, &acl_type);
-		if (r == 0) {
+		if (default_entry_acl_type & ARCHIVE_ENTRY_ACL_TYPE_NFS4) {
```

Comment:
There is no need to call acl_get_entry_type_np() and later acl_get_flagset_np() when we are not working on a NFSv4 ACL. They would return EINVAL on FreeBSD. This way we always expect the functions to return 0.